pkg/daemon/ceph/osd/daemon.go

```diff
@@ -154,15 +164,19 @@ func Run(context *clusterd.Context, agent *OsdAgent, done chan struct{}) error {
 	mon := NewMonitor(context, agent)
 	go mon.Run()
```

Comment:
can we move this monitoring of the osds to the operator? It doesn't seem to belong in this preparation pod anymore since we can manage at the deployment/pod level

Reply:
@travisn I think this still makes sense to have here because the monitoring here can directly work with if needed the admin socket and trigger a restart of just the OSD process, where the operator possibly could "just" delete the Pod.

Still a way for the operator to restart (e.g. delete the Pod) that seems completely "stuck" sounds good to me but for me this doesn't need to be done in this PR.

Reply:
@galexrt the preparation pod is only temporary. When the osd daemon pod is started, it will be started directly with the `ceph osd` command and rook won't have code running in that pod. So the only place to monitor osds would be in the operator